src/osd/OSD.cc

```diff
@@ -5678,6 +5678,9 @@ void OSD::_preboot(epoch_t oldest, epoch_t newest)
   // if our map within recent history, try to add ourselves to the osdmap.
   if (osdmap->get_epoch() == 0) {
     derr << "waiting for initial osdmap" << dendl;
+  } else if (osdmap->is_destroyed(whoami)) {
+    derr << "osdmap says I am destroyed, exiting" << dendl;
+    exit(1);
```

Comment:
Do you know what the rule is (if any) on how we should exit if we want systemd not to try to restart us?